build/source/driver/multi_driver.f90

```diff
@@ -827,32 +876,76 @@ program multi_driver
  ! *** model simulation
  ! ****************************************************************************
 
+ ! initialize the start of the physics
+ call date_and_time(values=startPhysics)
+
  ! ----- rank the GRUs in terms of their anticipated computational expense -----
 
- ! use persistence 
+ ! estimate computational expense based on persistence 
  !  -- assume that that expensive GRUs from a previous time step are also expensive in the current time step
 
- ! get the total number of flux calls from the previous time step (for all HRUs within a GRU)
- allocate(totalFluxCalls(nGRU), ixFluxCalls(nGRU), stat=err)
- call handle_err(err,'unable to allocate space for totalFluxCalls')
+ ! compute the total number of flux calls from the previous time step
  do jGRU=1,nGRU
   totalFluxCalls(jGRU) = 0._dp
   do iHRU=1,gru_struc(jGRU)%hruCount
    totalFluxCalls(jGRU) = totalFluxCalls(jGRU) + indxStruct%gru(jGRU)%hru(iHRU)%var(iLookINDEX%numberFluxCalc)%dat(1)
   end do
  end do
 
- ! get the indices that can rank the number of flux calls
- call indexx(totalFluxCalls, ixFluxCalls) ! ranking of each GRU w.r.t. computational expense
- ixFluxCalls=ixFluxCalls(nGRU:1:-1)       ! reverse ranking: now largest to smallest
+ ! get the indices that can rank the computational expense
+ !call indexx(totalFluxCalls, ixExpense) ! ranking of each GRU w.r.t. computational expense
+ call indexx(timeGRU, ixExpense) ! ranking of each GRU w.r.t. computational expense
+ ixExpense=ixExpense(nGRU:1:-1)  ! reverse ranking: now largest to smallest
+
+ ! initialize the GRU count
+ ! NOTE: this needs to be outside the parallel section so it is not reinitialized by different threads
+ kGRU=0
+
+ ! initialize the time that the openMP section starts
+ openMPstart = omp_get_wtime()
```

Comment:
@martynpclark - this is a OpenMP function. Unless I'm missing something, this will require the OpenMP compiler flag, correct?

Reply:
Yes, I was wondering the same thing. The code compiles and runs fine without the `openMP` flag. Do you have suggestions on any `if..then..else` statements to wrap such things (e.g., if openMP then use the function, if not use system clock)?

Reply:
Not sure how this would work in Fortran per se but this is what I did in VIC: 

https://github.com/jhamman/VIC/blob/74dc13e77d613b811fa889e1b6339c8aa04a070e/vic/drivers/shared_image/include/vic_mpi.h#L32-L36

I use the system clocks.

Reply:
It doesn't compile for me (`gfortran`)